code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -357,13 +357,15 @@
 		P.fire()
 		return P
 
+/mob/living/simple_animal/hostile/proc/CanSmashTurfs(turf/T)
+	return iswallturf(T)
```

Comment:
this is default functionality change.

The default, as shown below, was to allow both walls and mineral turfs to be destroyed.
Yes it's now overridable, but you've changed the default and we don't tend to do that in refactors without good justification.